To: vim_dev@googlegroups.com
Subject: Patch 7.3.167
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.167
Problem:    When using the internal grep QuickFixCmdPost is not triggered.
	    (Yukihiro Nakadaira)
Solution:   Change the place where autocommands are triggered.
Files:	    src/quickfix.c


*** ../vim-7.3.166/src/quickfix.c	2011-04-11 21:35:03.000000000 +0200
--- src/quickfix.c	2011-04-28 13:28:03.000000000 +0200
***************
*** 2742,2747 ****
--- 2742,2754 ----
  #ifdef FEAT_AUTOCMD
      char_u	*au_name = NULL;
  
+     /* Redirect ":grep" to ":vimgrep" if 'grepprg' is "internal". */
+     if (grep_internal(eap->cmdidx))
+     {
+ 	ex_vimgrep(eap);
+ 	return;
+     }
+ 
      switch (eap->cmdidx)
      {
  	case CMD_make:	    au_name = (char_u *)"make"; break;
***************
*** 2763,2775 ****
      }
  #endif
  
-     /* Redirect ":grep" to ":vimgrep" if 'grepprg' is "internal". */
-     if (grep_internal(eap->cmdidx))
-     {
- 	ex_vimgrep(eap);
- 	return;
-     }
- 
      if (eap->cmdidx == CMD_lmake || eap->cmdidx == CMD_lgrep
  	|| eap->cmdidx == CMD_lgrepadd)
  	wp = curwin;
--- 2770,2775 ----
***************
*** 3057,3066 ****
  
      switch (eap->cmdidx)
      {
! 	case CMD_vimgrep: au_name = (char_u *)"vimgrep"; break;
! 	case CMD_lvimgrep: au_name = (char_u *)"lvimgrep"; break;
! 	case CMD_vimgrepadd: au_name = (char_u *)"vimgrepadd"; break;
  	case CMD_lvimgrepadd: au_name = (char_u *)"lvimgrepadd"; break;
  	default: break;
      }
      if (au_name != NULL)
--- 3057,3070 ----
  
      switch (eap->cmdidx)
      {
! 	case CMD_vimgrep:     au_name = (char_u *)"vimgrep"; break;
! 	case CMD_lvimgrep:    au_name = (char_u *)"lvimgrep"; break;
! 	case CMD_vimgrepadd:  au_name = (char_u *)"vimgrepadd"; break;
  	case CMD_lvimgrepadd: au_name = (char_u *)"lvimgrepadd"; break;
+ 	case CMD_grep:	      au_name = (char_u *)"grep"; break;
+ 	case CMD_lgrep:	      au_name = (char_u *)"lgrep"; break;
+ 	case CMD_grepadd:     au_name = (char_u *)"grepadd"; break;
+ 	case CMD_lgrepadd:    au_name = (char_u *)"lgrepadd"; break;
  	default: break;
      }
      if (au_name != NULL)
*** ../vim-7.3.166/src/version.c	2011-04-28 17:24:54.000000000 +0200
--- src/version.c	2011-04-28 17:26:17.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     167,
  /**/

-- 
Micro$oft: where do you want to go today?
    Linux: where do you want to go tomorrow?
  FreeBSD: are you guys coming, or what?

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
